app/controllers/collection_participants_controller.rb

```diff
@@ -54,29 +55,29 @@ def join
     end
     participants = CollectionParticipant.in_collection(@collection).for_user(current_user) unless current_user.nil?
     if participants.empty?
-      @participant = CollectionParticipant.new(:collection => @collection, :pseud => current_user.default_pseud, 
+      @participant = CollectionParticipant.new(:collection => @collection, :pseud => current_user.default_pseud,
                         :participant_role => CollectionParticipant::NONE)
       @participant.save
       flash[:notice] = t('applied_to_join_collection', :default => "You have applied to join %{collection}.", :collection => @collection.title)
     else
       participants.each do |participant|
         if participant.is_invited?
-          participant approve_membership!
```

Comment:
I don't believe this previously worked; if someone else has more background, please let me know!